### PR TITLE
done with like, history & watch later

### PR DESCRIPTION
**Pull request by Varun-28**
###Done with watch-later, history and likes feat

*Files to review:*

- userData-context.js
- All the serverCall files
- Single Video.js

*Features Added:*

- Automatic add to the history
- remove and clear history
- add and remove from likes
- add and remove from watchlater
- validation that user is login or not

### CODE REVIEW

File: src/components/CommonCard.jsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        <button
+          className="card-btn-icon"
+          onClick={() => {
+            if (type === "history") {
+              removeFromHistory(_id, dataDispatch);
+            } else if (type === "like") {
+              removeFromLikes(_id, dataDispatch);
+            } else if (type === "watchlater") {
+              removeFromWatchLater(_id, dataDispatch);
+            }
```
Comment (abhij1607) [inline src/components/CommonCard.jsx:49]: This can be abstracted to separate function to make it more readable

**Review by abhij1607 — APPROVED** [review]
Liked the work with common card
Logic looks good
Can be merged

File: src/components/CommonCard.jsx (hunk 1 above)
Comment (Varun-28) [reply src/components/CommonCard.jsx:49]: Thanks for the suggestion, I have changed it in the next branch

File: src/context/userDataContext/watchLater-serverCalls.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+const getWatchlater = async (dataDispatch) => {
+  const token = localStorage.getItem("userToken");
+  const {
+    data: { watchlater },
+  } = await axios.get("/api/user/watchlater", {
+    headers: { authorization: token },
+  });
+  dataDispatch({ type: "WATCHLATER", payload: watchlater });
+};
```
Comment (Tanay-J) [inline src/context/userDataContext/watchLater-serverCalls.js:11]: consider wrapping api calls in a try catch block

**Review by Tanay-J — APPROVED** [review]
Clean code. Liked separation various api calls in different files. You may consider refactoring a bit for the suggestion I have made.

File: src/context/userDataContext/watchLater-serverCalls.js (hunk 2 above)
Comment (Varun-28) [reply src/context/userDataContext/watchLater-serverCalls.js:11]: Ooh thanks, I think I missed it here!
I will correct it in the next branch 🤯

**Pull request merged**